dlt/destinations/snowflake/snowflake.py

```diff
@@ -48,44 +53,82 @@
     "VARIANT": "complex"
 }
 
+@with_config(spec=AwsCredentials, sections=(known_sections.DESTINATION, "filesystem", "credentials"))
+def _s3_config(config: AwsCredentials = config.value) -> AwsCredentials:
+    return config
+
+
 class SnowflakeLoadJob(LoadJob, FollowupJob):
     def __init__(
             self, file_path: str, table_name: str, write_disposition: TWriteDisposition, load_id: str, client: SnowflakeSqlClient,
-            stage_name: Optional[str] = None, keep_staged_files: bool = True
+            stage_name: Optional[str] = None, storage_integration: Optional[str] = None, keep_staged_files: bool = True
     ) -> None:
         file_name = FileStorage.get_file_name_from_file_path(file_path)
         super().__init__(file_name)
 
-        with client.with_staging_dataset(write_disposition == "merge"):
+        with client.with_staging_dataset(write_disposition in ["merge", "replace"]):
             qualified_table_name = client.make_qualified_table_name(table_name)
 
-            if stage_name:
```

Comment:
let's change how `stage_name` works:
1. if stage name is specified we expect that this stage is created and contains all expected integrations and credentials so we just pass it in copy command. we also need to use a pattern that will load only this specific file
2. if stage name is not provided then:
- for local files that use PUT we use table stage (like we do now)
- for bigquery we just simply require stage name. if someone can create integration then s/he can create stage as well
- for s3/azure (we'll add soon) we can pass credentials in copy command as you do, just see comment below

Reply:
if the above works we do not need to create any stages and we do not need `storage_integration`